core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -123,6 +123,10 @@ class NewHadoopRDD[K, V](
 
   override def getPartitions: Array[Partition] = {
     val inputFormat = inputFormatClass.getConstructor().newInstance()
+    // setMinPartitions below will call FileInputFormat.listStatus(), which can be quite slow when
+    // traversing a large number of directories and files. Parallelize it.
+    _conf.setIfUnset(FileInputFormat.LIST_STATUS_NUM_THREADS,
+      Runtime.getRuntime.availableProcessors().toString)
```

Comment:
maybe we shall revamp the value for k8s?

Reply:
Can you refer to the existing code if we have done this before?

Reply:
yes, https://github.com/apache/spark/blob/2d085c13b7f715dbff23dd1f81af45ff903d1a79/core/src/main/scala/org/apache/spark/rdd/WholeTextFileRDD.scala#L45-L46

https://github.com/apache/spark/blob/2d085c13b7f715dbff23dd1f81af45ff903d1a79/core/src/main/scala/org/apache/spark/rdd/BinaryFileRDD.scala#L42-L43

Reply:
Will this `availableProcessors` get the actual number of available CPU cores in container instances, VM instances and physical machine instances?

Reply:
I'm not sure if using `Runtime.getRuntime.availableProcessors()` is really reasonable

Reply:
yes, as I mentioned https://github.com/apache/spark/pull/34792#discussion_r761702407, maybe some magic number 8/16 works better for all scenarios if we choose to hardcode it